File: Python/quiz.py

def main():
    questions = [ 
        {
            'q': "In what year was Python 1.0 released?", 
            'a': "1991"
        }, {
            'q': "In what year was Python 2.0 released?",
            'a': "2000"
        }, {
            'q': "In what year was Python 3.0 released?",
            'a': "2008"
        }, {
            'q': "Who is the creator of Python?",
            'a': "Guido van Rossum"
        }, {
            'q': "What language was Python designed as a successor to?",
            'a': "ABC"
        }, {
            'q': "How many reserved keywords does Python have?",
            'a': "33"
        }, {
            'q': "How do you denote a single-line comment in Python?",
            'a': "#"
        }, {
            'q': "What is the standard typing style for variables in Python?",
            'a': "snake_case"
        }
    ]

    incorrect_guesses = 0

    for question in questions:
        print(question['q'])

        guess = ""
        while guess != question['a'] and guess != "skip":
            guess = input("> ")

            if guess == "skip":
                incorrect_guesses += 10
            elif guess != question['a']:
                incorrect_guesses += 1
            else:
                print("correct")
                print()

    print("Incorrect Guesses: " + str(incorrect_guesses))

if __name__ == "__main__":
    main()

File: C/quiz.c
#include <stdio.h>
#include <string.h>

typedef struct {
    char *q;
    char *a;
} question_answer;

int main() {
    question_answer questions[] = {
        {"In what year was C first released?", "1972"},
        {"Who was the lead designer of C?", "Dennis Ritchie"},
        {"Who co-authoured The C Programming Language with the lead designer?", "Brian Kernighan"},
        {"How many reserved keywords does C have?", "32"},
        {"What is the standard typing style for variables in C?", "snake_case"},
    };

    int incorrect_guesses = 0;

    for (int i = 0; i < sizeof(questions)/sizeof(questions[0]); i++) {
        printf("%s\n", (*(questions+i)).q);

        char guess[50] = "";
        while (strcmp(guess, (*(questions+i)).a) != 0 && strcmp(guess, "skip") != 0) {
            printf("> ");
            fgets(guess, 50, stdin);
            guess[strcspn(guess, "\n")] = 0;

            if (strcmp(guess, "skip") == 0) {
                incorrect_guesses += 10;
                printf("%s\n\n", (*(questions+i)).a);
            } else if (strcmp(guess, (*(questions+i)).a)) {
                incorrect_guesses++;
            } else {
                printf("correct\n\n");
            }
        }
    }

    printf("Incorrect Guesses: %d\n", incorrect_guesses);
}

